core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiServiceSpec.scala

```diff
@@ -142,6 +225,62 @@ class SubmissionApiServiceSpec extends ApiServiceSpec {
     }
   }
 
+  it should "abort a submission" in withTestDataApiServices { services =>
+    val wsName = testData.wsName
+    val mcName = MethodConfigurationName("no_input", "dsde", wsName)
+    val methodConf = MethodConfiguration(mcName.namespace, mcName.name, "Sample", Map.empty, Map.empty, Map.empty, MethodRepoMethod("dsde", "no_input", 1))
+
+    // create a submission
+    val submission = createAndMonitorSubmission(wsName, methodConf, testData.sample1, None, services, Some(WorkflowFailureModes.ContinueWhilePossible.toString))
+
+    assertResult(1) {
+      submission.workflows.size
+    }
+
+    abortSubmission(services, wsName, submission.submissionId)
+  }
+
+  it should "create and abort a large submission" in withLargeSubmissionApiServices { services =>
+    val wsName = testData.wsLargeSubmission
+    val mcName = MethodConfigurationName("no_input", "dsde", wsName)
+    val methodConf = MethodConfiguration(mcName.namespace, mcName.name, "Sample", Map.empty, Map.empty, Map.empty, MethodRepoMethod("dsde", "no_input", 1))
+
+    // create a submission
+    val submission = createAndMonitorSubmission(wsName, methodConf, testData.largeSset, Option("this.hasSamples"), services)
+
+    assertResult(10000) {
+      submission.workflows.size
+    }
+
+    abortSubmission(services, wsName, submission.submissionId)
+  }
+
+  // Ignored as this test is prone to intermittent timeout-related failures.
```

Comment:
Hm... I'll check again. I was seeing timeout errors (the test API timeout is set to 5 sec) due to the sheer number of workflows, not any database deadlock. Maybe I could increase the timeout for this test only to get it to pass consistently.